src/toxcore/stats.rs

```diff
@@ -0,0 +1,110 @@
+/*!
+Statistics of incoming/outgoing packets
+This is used by both Udp codec and Tcp codec.
+*/
+
+#[cfg(target_pointer_width = "64")]
+use std::sync::atomic::*;
+#[cfg(target_pointer_width = "32")]
+use std::sync::Mutex;
+use std::sync::Arc;
+
+#[derive(Clone, Debug, Default)]
+/// Struct for counting packets on x64 CPU.
+#[cfg(target_pointer_width = "64")]
+pub struct Stats {
+    /// Incoming packets count for Udp/Tcp
+    incoming: Arc<AtomicUsize>,
+    /// Outgoing packets count for Udp/Tcp
+    outgoing: Arc<AtomicUsize>,
+}
+
+#[derive(Clone, Debug, Default)]
+/// Struct for counting packets on x86 CPU.
+#[cfg(target_pointer_width = "32")]
+pub struct Stats {
+    /// Incoming packets count for Udp/Tcp
+    pub incoming: Arc<Mutex<u64>>,
+    /// Outgoing packets count for Udp/Tcp
+    pub outgoing: Arc<Mutex<u64>>,
+}
+
+impl Stats {
+    /// New Stats object
+    pub fn new() -> Self {
+        Default::default()
+    }
+
+    /// Add 1 to incoming counter
+    #[cfg(target_pointer_width = "64")]
+    pub fn increase_incoming(&self) {
+        self.incoming.fetch_add(1, Ordering::SeqCst);
+    }
+
+    /// Add 1 to incoming counter
+    #[cfg(target_pointer_width = "32")]
+    pub fn increase_incoming(&self) {
+        *self.incoming.lock().expect("Can't lock mutex") += 1;
+    }
+
+    /// Add 1 to incoming counter
+    #[cfg(target_pointer_width = "64")]
+    pub fn increase_outgoing(&self) {
+        self.outgoing.fetch_add(1, Ordering::SeqCst);
+    }
+
+    /// Add 1 to incoming counter
+    #[cfg(target_pointer_width = "32")]
+    pub fn increase_outgoing(&self) {
+        *self.outgoing.lock().expect("Can't lock mutex") += 1;
+    }
+
+    /// Get incoming count
```

Comment:
In `increase_*` docs you say `Add 1 to incoming counter` whereas in `get_*` docs you say `count`. There should be 1 style

Reply:
done